core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -1130,6 +1134,24 @@ class SocketServerTest {
       TestUtils.waitUntilTrue(() => totalTimeHistCount() == expectedTotalTimeCount,
         s"request metrics not updated, expected: $expectedTotalTimeCount, actual: ${totalTimeHistCount()}")
 
+      TestUtils.waitUntilTrue(() => {
+        val updatedTotalTimeBucketHistCount = getTotalTimeBucketHistCount()
+        assertTrue(updatedTotalTimeBucketHistCount.size > 0)
```

Comment:
I think the waitUntilTrue shouldn't include assertTrue inside, since a failure of the assertion may abort the whole test.

Reply:
Those asserts should be met regardlessly. If any of the asserts fail, the test should fail. The only wait condition is "increaseCount == 1".